ui/store/actions.js

```diff
@@ -3110,6 +3119,170 @@ export async function setWeb3ShimUsageAlertDismissed(origin) {
   await promisifiedBackground.setWeb3ShimUsageAlertDismissed(origin);
 }
 
+// Smart Transactions Controller
+export async function setSmartTransactionsOptInStatus(optInState) {
+  await promisifiedBackground.setSmartTransactionsOptInStatus(optInState);
+}
+
+export function fetchSmartTransactionFees(unsignedTransaction) {
+  return async (dispatch) => {
+    try {
+      const smartTransactionFees = await promisifiedBackground.fetchSmartTransactionFees(
+        unsignedTransaction,
+      );
+      dispatch({
```

Comment:
Do we need to add this to redux state? Could we just add it to `state.metamask.smartTransactionsState` by setting it on state in the controller?

Reply:
Sure, we can do that.